src/App.js

```diff
@@ -14,11 +14,87 @@ const goodsFromServer = [
   'Garlic',
 ];
 
-const App = () => (
-  <div className="App">
-    <h1>Selected good: -</h1>
-    {goodsFromServer.length}
-  </div>
-);
+class App extends Component {
+  state = {
+    selectedGoods: ['Jam'],
+  };
+
+  render() {
+    const { selectedGoods } = this.state;
+
+    function goodsSelector(goods) {
+      switch (goods.length) {
+        case 0:
+          return 'No goods selected';
+        case 1:
+          return `${goods[0]} is selected`;
+        default:
+          return `${goods.slice(0, goods.length - 1).join(', ')}`
+          + ` and ${goods[goods.length - 1]} are selected`;
+      }
+    }
+
+    const removeGood = (goods, good) => {
+      this.setState({ selectedGoods: goods
+        .filter(product => product !== good) });
+    };
+
+    const addGood = (goods, good) => {
+      this.setState({ selectedGoods: [...goods, good] });
+    };
+
+    const clearAllGoods = () => {
+      this.setState({ selectedGoods: [] });
+    };
+
+    return (
+      <div className="App">
+        <div className="App__wrapper">
+          <h1 className="App__header">
+            {goodsSelector(selectedGoods)}
+          </h1>
+          <button
+            type="button"
+            className={
+              selectedGoods.length === 0
+                ? 'App__clear-button--disable'
+                : 'App__clear-button'
```

Comment:
I think it's time to start using classnames lib ;)
https://www.npmjs.com/package/classnames